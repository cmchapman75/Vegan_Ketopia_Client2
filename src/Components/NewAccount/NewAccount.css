.account-creation {
    text-align: center;

}

.creation-form {
    padding: 1px;
}

.field-label {
     
    /* margin: 2px; */
}

.field-input {
    padding: 0 1px;
}

.medButton {
    padding: 8px;
    margin: 5px;
    border-radius: 25%;
    color: white;
    background-color: rgb(2, 61, 2);
}