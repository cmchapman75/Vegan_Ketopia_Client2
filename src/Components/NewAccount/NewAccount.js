import React, { Component } from 'react';
import { Link } from "react-router-dom";

import AuthHelper from "../../Helpers/Auth";

import "./NewAccount.css";

class NewAccount extends Component {
    constructor(props) {
        super(props);
        this.state ={
            error: null,            
            username: '',
            email_address: '',
            password: ''
        }
    
        this.handleChange = this.handleChange.bind(this);
        this.createSubmit = this.createSubmit.bind(this);
      }
    // static defaultProps = {
    //     location: {},
    //     history: {
    //         push: () => { }
    //     },        
    // };
    
    state = { error: null};
    handleChange(event) {
        this.setState({[event.target.name]: event.target.value});
    }

    handleCreationSuccess = () => {
        const { history } = this.props;
        history.push("/login");
    };

    

    createSubmit = e => {
        e.preventDefault();
        const { username, email_address, password } = e.target;
        console.log(username, email_address, password);

        this.setState({ error: null });

        AuthHelper.createUser({
            username: username.value,
            email_address: email_address.value,
            password: password.value
        })
        .then(user => {
            username.value = "";
            email_address.value = "";
            password.value = "";
            this.handleCreationSuccess();
        })
        .catch(res => {
            this.setState({ error: res.error });
        });
    };

    // componentDidMount() {
    //     this.
    // }
    render() {
        let error = this.state.error;
        return(
            <div className="account-creation">
                <header className="creation-header"></header>
                <form className="creation-form" onSubmit={this.createSubmit}>
                    <div role={"alert"}>
                        {error && <p className="red" id="error-message">{this.state.error}</p>}                        
                    </div>
                    <label className="field-label">
                        Username:
                    </label>
                    <input
                        className="field-input"
                        required
                        name="username"
                        value={this.state.username}
                        onChange={e => this.handleChange(e)}
                        // placeholder="Username"
                    />
                    <label className="field-label">
                        Email Address:
                    </label>
                    <input 
                        className="field-input"
                        required
                        name="email_address"
                        value={this.state.email_address}
                        onChange={e => this.handleChange(e)}
                        // placeholder="Email Address"
                    />  
                    <label className="field-label">
                        Password:    
                    </label>  
                    <input 
                        className="field-input"
                        required
                        name="password"
                        value={this.state.password}
                        onChange={e => this.handleChange(e)}
                        // placeholder="Password"
                    />  
                    <div className="button-con">
                        <button className="medButton" id="create-button">Create</button>
                        <br />
                        <Link to="/Login">
                            <button className="medButton">Already a Ketopian?</button>
                        </Link>                    
                    
                    </div>  

                </form>
            </div>
        )
    }

}


export default NewAccount;