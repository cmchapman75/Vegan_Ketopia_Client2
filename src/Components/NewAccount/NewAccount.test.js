import React from 'react';
import ReactDOM from 'react-dom';
import NewAccount from './NewAccount';
import { BrowserRouter } from 'react-router-dom';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter>
    <NewAccount />
  </BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<BrowserRouter>
    <NewAccount />
  </BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});