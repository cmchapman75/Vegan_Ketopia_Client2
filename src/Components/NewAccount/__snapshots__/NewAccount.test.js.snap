// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="account-creation"
>
  <header
    className="creation-header"
  />
  <form
    className="creation-form"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <label
      className="field-label"
    >
      Username:
    </label>
    <input
      className="field-input"
      name="username"
      onChange={[Function]}
      required={true}
      value=""
    />
    <label
      className="field-label"
    >
      Email Address:
    </label>
    <input
      className="field-input"
      name="email_address"
      onChange={[Function]}
      required={true}
      value=""
    />
    <label
      className="field-label"
    >
      Password:
    </label>
    <input
      className="field-input"
      name="password"
      onChange={[Function]}
      required={true}
      value=""
    />
    <div
      className="button-con"
    >
      <button
        className="medButton"
        id="create-button"
      >
        Create
      </button>
      <br />
      <a
        href="/Login"
        onClick={[Function]}
      >
        <button
          className="medButton"
        >
          Already a Ketopian?
        </button>
      </a>
    </div>
  </form>
</div>
`;
